src/parser.rs

```diff
@@ -2180,3 +2171,71 @@ where
         self.fix_position(err)
     }
 }
+
+impl<'de, R> Parser<R>
+where
+    R: Reader<'de>,
+{
+    pub fn get_by_schema(&mut self, schema: &mut crate::Value) -> Result<()> {
+        let mut strbuf = Vec::with_capacity(DEFAULT_KEY_BUF_CAPACITY);
+        self.get_by_schema_rec(schema, &mut strbuf)
+    }
+
+    fn get_by_schema_rec(&mut self, schema: &mut crate::Value, strbuf: &mut Vec<u8>) -> Result<()> {
+        let ch = self.skip_space_peek();
+        if ch.is_none() {
+            return perr!(self, EofWhileParsing);
+        }
+
+        let mut should_change = true;
+        let start = self.read.index();
+
+        match (schema.as_object_mut(), self.skip_space_peek()) {
+            (Some(object), Some(b'{')) => {
+                // deal with the empty object
+                match self.get_next_token([b'"', b'}'], 1) {
+                    Some(b'"') => {}
+                    Some(b'}') => return Ok(()),
+                    None => return perr!(self, EofWhileParsing),
+                    Some(_) => unreachable!(),
+                }
+
+                let mut key_values = HashMap::new();
+                for (key, value) in object.iter_mut() {
+                    key_values.insert(key, value);
+                }
+
+                loop {
+                    let key = self.parse_str_impl(strbuf)?;
+                    self.parse_object_clo()?;
+                    if let Some(val) = key_values.get_mut(key.deref()) {
+                        should_change = false;
+                        self.get_by_schema_rec(val, strbuf)?;
+                    } else {
+                        self.skip_one()?;
+                    }
+
+                    match self.skip_space() {
+                        Some(b',') => match self.skip_space() {
+                            Some(b'"') => continue,
+                            _ => return perr!(self, ExpectObjectKeyOrEnd),
+                        },
+                        Some(b'}') => break,
+                        Some(_) => return perr!(self, ExpectedObjectCommaOrEnd),
+                        None => return perr!(self, EofWhileParsing),
+                    }
+                }
+            }
+            _ => {
+                self.skip_one()?;
+            }
+        }
+
+        let end = self.read.index();
```

Comment:
`should_change` means `has_found` ? If not found, maybe not need to override the `schema` Value

Reply:
My fault. I add two more test cases from sonic-cpp. I find an object with the empty value `{}`, which is regarded as a default value rather than a nested query, can be replaced in its entirety. But when it has some members, it is regarded as a nested query and will never be replaced.